EIPS/eip-00.md

```diff
@@ -0,0 +1,789 @@
+requires to complete successfully while not having the actual `signature` value, this function
+should avoid reverting on invalid signature, and should return a value different from `MAGIC_VALUE_SENDER`.
+
+The `validateTransaction` function can choose to revert on any condition that can be satisfied during gas estimation.
+
+#### Paymaster validation frame
+
+The `paymaster` of the transaction, if specified, is invoked with the following data:
+
+```solidity
+
+function validatePaymasterTransaction(uint256 version, bytes32 txHash, bytes transaction) external returns (bytes context, uint256 validationData);
+
+```
+
+The gas limit of this frame is set to `paymasterGasLimit`.
+
+The amount of gas used by this frame is referred to as `paymasterValidationGasUsed`.
+
+The `transaction` parameter is interpreted as an ABI encoding of `TransactionType4`.
+
+The frame must return a bytes array that is interpreted as:
+
+```solidity
+
+abi.encode(context, MAGIC_VALUE_PAYMASTER, validBefore, validAfter)
+
+```
+
+Same as in the [`sender` validation frame](#sender-validation-frame), in order to support gas estimation this
+frame should return a value different from `MAGIC_VALUE_PAYMASTER` for conditions that cannot be satisfied
+before signing.
+
+The size of the `context` byte array may not exceed `MAX_CONTEXT_SIZE` for a transaction to be considered valid.
+
+#### Sender execution frame
+
+The `sender` address is invoked with `callData` input.
+
+The gas limit of this frame is set to `callGasLimit`.
+The amount of gas used by this frame is referred to as `gasUsedByExecution`.
+
+The validation frames do not revert if the execution frame reverts.
+The `postPaymasterTransaction` may still be called with a `success: false` flag.
+
+#### Paymaster post-transaction frame
+
+After the sender execution frame is over the `paymaster` may need to perform some kind of cleanup or bookkeeping.
+If the gas payment validation returned a non-zero `context`, the `paymaster` is invoked again
+with the following inputs:
+
+```solidity
+
+function postPaymasterTransaction(bool success, uint256 actualGasCost, bytes context) external;
+
+```
+
+The `actualGasCost` parameter is the actual amount paid by the paymaster for this transaction,
+and `success` indicates whether this transaction's execution frame completed without revert.
+
+The gas limit of this frame is set to `paymasterGasLimit - paymasterValidationGasUsed`.
+
+Revert in the `postPaymasterTransaction` frame reverts the transaction's execution frame as well.
+The validation frames do not revert if the `postPaymasterTransaction` frame reverts.
+The gas fees charged from the `paymaster` will still include the gas cost of the reverted execution frame.
+
+### Execution layer transaction validation
+
+On the execution layer, the transaction validity conditions for a block are extended as follows:
+
+```
+
+func validateAccountAbstractionTransaction(tx *Transaction) {
+
```

Comment:
assert ! (sender.code.length > 0 && deployerData.length > 0)

Reply:
Right.  We should add that.

Reply:
> assert ! (sender.code.length > 0 && deployerData.length > 0)

That's right, added this check. Thanks!